### ISSUE TITLE
Temperature=1.0 does not generate the same output

### ISSUE THREAD

**Issue by dkobak** [karpathy/nanoGPT#233]
When sampling with `temperature=1.0`, I expect to get the same results every time. That's what the comment in `sample.py` implies:
```
temperature = 0.8 # 1.0 = no change, < 1.0 = less random, > 1.0 = more random, in predictions
```

However, this does not happen to me. Here is an example. I train the model via
```
python train.py config/train_shakespeare_char.py
```
(this bring me to val loss 1.4809 at iter 1750, afterwards the validation loss starts increasing).
Then sampling as follows:
```
python sample.py --out_dir=out-shakespeare-char --num_samples=5 --start=$'Tomorrow' --temperature=1.0 --max_new_tokens=30
```
gives me this:
```
Tomorrow?

CAMILLO:
My wife, is not th
---------------
Tomorrow
That way before.
Stirr'd Merc
---------------
Tomorrow, ha!

FLORIZEL:
How now, that
---------------
Tomorrow dilth and swim, what fast,
An
---------------
Tomorrow or me
Yours, to use her to sl
---------------
```

Different sample every time, despite temperature set to 1.0.

**Comment by dkobak** [comment]
Okay I have to close this immediately. One needs low temperature to get deterministic results, not high temperature :-/

With `--temperature=0.001` I get 

```
Tomorrow the season of the seas
Of the
---------------
Tomorrow the season of the seas
Of the
---------------
Tomorrow the season of the seas
Of the
---------------
Tomorrow the season of the seas
Of the
---------------
Tomorrow the season of the seas
Of the
---------------
```

Apologies for raising this.
